src/alt.js

```diff
@@ -79,8 +79,12 @@ class AltStore {
       if (model[LISTENERS][payload.action]) {
         this[SET_STATE] = false
         const result = model[LISTENERS][payload.action](payload.data)
-        if (result !== false && this[SET_STATE] === false) {
-          this.emitChange()
+        if (this[SET_STATE] === false) {
+          if (result instanceof Promise) {
```

Comment:
I know this is an old commit that doesn't seem that it'll get merged it. Just an FYI that `object instanceof Promise` fails on Chrome for native promises.